packages/web3-eth-contract/src/index.js

```diff
@@ -56,22 +56,24 @@ var Contract = function Contract(jsonInterface, address, options) {
         args = Array.prototype.slice.call(arguments);
 
     if(!(this instanceof Contract)) {
-        throw new Error('Please use the "new" keyword to instantiate a web3.eth.contract() object!');
+        throw new Error('Please use the "new" keyword to instantiate a web3.eth.Contract() object!');
     }
 
-    // sets _requestmanager
-    core.packageInit(this, [this.constructor.currentProvider]);
+    this.setProvider = function () {
+        core.packageInit(_this, arguments);
 
-    this.clearSubscriptions = this._requestManager.clearSubscriptions;
+        _this.clearSubscriptions = _this._requestManager.clearSubscriptions;
+    };
```

Comment:
@nivida Could you eli5 me what this setProvider definition means and how it's different than Contract.setProvider? 

In Contract.setProvider it's possible to set accounts for example...

https://github.com/ethereum/web3.js/blob/a46dbcd7bfa6a0bb4c69e0ededcdb0646321c741/packages/web3-eth-contract/src/index.js#L241-L246

Reply:
This is the static ``setProvider`` method of the ``Contract`` object which will be [used from the ``web3-eth``](https://github.com/ethereum/web3.js/blob/1.x/packages/web3-eth/src/index.js#L304) module to pre-set the ``provider`` and the ``web3-eth-accounts`` for the initiation of the class later. This ``setProvider`` method is also documented as usage example in the [readme of the ``web3-eth-contract``](https://github.com/ethereum/web3.js/tree/1.x/packages/web3-eth-contract#usage) package.